indexing-service/src/main/java/io/druid/indexing/common/task/RealtimeIndexTask.java

```diff
@@ -366,33 +372,35 @@ public String getVersion(final Interval interval)
     finally {
       if (normalExit) {
         try {
-          // Always want to persist.
-          log.info("Persisting remaining data.");
-
-          final Committer committer = committerSupplier.get();
-          final CountDownLatch persistLatch = new CountDownLatch(1);
-          plumber.persist(
-              new Committer()
-              {
-                @Override
-                public Object getMetadata()
-                {
-                  return committer.getMetadata();
-                }
-
-                @Override
-                public void run()
+          // Persist if we had actually started.
+          if (firehose != null) {
+            log.info("Persisting remaining data.");
+
+            final Committer committer = committerSupplier.get();
+            final CountDownLatch persistLatch = new CountDownLatch(1);
+            plumber.persist(
+                new Committer()
                 {
-                  try {
-                    committer.run();
+                  @Override
+                  public Object getMetadata()
+                  {
+                    return committer.getMetadata();
                   }
-                  finally {
-                    persistLatch.countDown();
+
+                  @Override
+                  public void run()
+                  {
+                    try {
+                      committer.run();
+                    }
+                    finally {
+                      persistLatch.countDown();
+                    }
                   }
                 }
-              }
-          );
-          persistLatch.await();
+            );
+            persistLatch.await();
```

Comment:
There's timeouts at higher levels (the thread pool task runner will interrupt this thread if it takes longer than the graceful shutdown timeout)